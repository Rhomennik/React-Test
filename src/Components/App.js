import React from 'react';
import Task from './Task';
// import { Container } from './styles';

const App = () => (
  <>
    <Task />
  </>
);

export default App;
