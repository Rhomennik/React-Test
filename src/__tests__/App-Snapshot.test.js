import React from "react";
import App from "../Components/App";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

it("should render the homepage for the non logged in user", () => {
  const props = {
    isLoggedIn: false,
    username: null
  };
  const app = mount(<App {...props} />);
  expect(toJson(app)).toMatchSnapshot();
});

it("should render the Homepage for the logged in user (Heisenberg)", () => {
  const props = {
    isLoggedIn: true,
    username: "Rhomennik"
  };
  const app = mount(<App {...props} />);
  expect(toJson(app)).toMatchSnapshot();
});

it("should render the Homepage for the logged in user without tasks", () => {
  const props = {
    isLoggedIn: true,
    username: "Rhome"
  };
  const app = mount(<App {...props} />);
  expect(toJson(app)).toMatchSnapshot();
});
