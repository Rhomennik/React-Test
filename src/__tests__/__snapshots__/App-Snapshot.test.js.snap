// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Homepage for the logged in user (Heisenberg) 1`] = `
<App
  isLoggedIn={true}
  username="Rhomennik"
>
  <div
    className="App"
  >
    <Home
      tasks={
        Array [
          "8am - Adonisjs",
          "12:30pm - Reactjs",
          "15pm - Graphql",
          "20pm - Apollo client & Server",
        ]
      }
      username="Rhomennik"
    >
      <div>
        <p>
          Welcome, 
          Rhomennik
          !
        </p>
        <ul>
          <li
            key="0"
          >
            8am - Adonisjs
          </li>
          <li
            key="1"
          >
            12:30pm - Reactjs
          </li>
          <li
            key="2"
          >
            15pm - Graphql
          </li>
          <li
            key="3"
          >
            20pm - Apollo client & Server
          </li>
        </ul>
      </div>
    </Home>
  </div>
</App>
`;

exports[`should render the Homepage for the logged in user without tasks 1`] = `
<App
  isLoggedIn={true}
  username="Rhome"
>
  <div
    className="App"
  >
    <Home
      tasks={
        Array [
          "8am - Rever Tarefas (typora)",
          "15pm - Comer ;)",
        ]
      }
      username="Rhome"
    >
      <div>
        <p>
          Welcome, 
          Rhome
          !
        </p>
        <ul>
          <li
            key="0"
          >
            8am - Rever Tarefas (typora)
          </li>
          <li
            key="1"
          >
            15pm - Comer ;)
          </li>
        </ul>
      </div>
    </Home>
  </div>
</App>
`;

exports[`should render the homepage for the non logged in user 1`] = `
<App
  isLoggedIn={false}
  username={null}
>
  <div
    className="App"
  >
    <p>
      Hello, visitor. Sign in to continue.
    </p>
  </div>
</App>
`;
